t/scripts/hydra-eval-jobset/notifications.t

```diff
@@ -69,7 +69,23 @@ subtest "on a fresh evaluation with changed sources" => sub {
     ok(evalSucceeds($builds->{"variable-job"}->jobset), "evaluating for the third time");
     is($listener->block_for_messages(0)->()->{"channel"}, "eval_started", "the evaluation started");
 
-    is($listener->block_for_messages(0)->()->{"channel"}, "build_queued", "expect only one new build being queued");
+    # The order of builds is randomized when writing to the database,
+    # so we can't expect the list in any specific order here.
+    is(
+        [sort(
+            $listener->block_for_messages(0)->()->{"channel"},
+            $listener->block_for_messages(0)->()->{"channel"},
+            $listener->block_for_messages(0)->()->{"channel"},
+            $listener->block_for_messages(0)->()->{"channel"}
+        )],
+        [
+            "build_queued",
```

Comment:
Would it be useful to document which job each of these events corresponds to? I imagine the two cached finished builds are both the stable passing and failing jobs. Is the cached queued build for the variable job? Or is that the queued build?

Reply:
Hmmm... yeah, I think that would be a good idea! It is a bit tricky because the events for builds are sent (on purpose by the hydra-eval-jobset code) in a randomized order, but we could piece it together here.

Reply:
I added comments about what the events correspond to. Writing the test to actually validate the content and deal with random order is, uh, frustrating.

Reply:
Much better and clearer, thanks!